docs/proposals/localDisk_volume.md

```diff
@@ -0,0 +1,309 @@
+# localDisk volume
+
+## Abstract
+
+This is a proposal for implementing a new volume type, localDisk, which will support,
+
+ 1. the Node has multiple physical disks and pod requires 1 ~ N disks as its storage volumes.
+ 2. the Node has lvm VG and pods allocate specific size LVs from this VG.
+ 3. When pods are deleted, the data in the volume could be either retained or deleted.
+ 
+As localDisk volume life cycle would be different from pod, we need volume APIs to do the management. 
+The existing PersistentVolume/PersistentVolumeClaim(PV/PVC) API is for network storage in cluster, we propose new LocalVolume/LocalVolumeClaim(LV/LVC) API to manage localDisk volume.                      
+
+## Motivation
+
+* This new volume type is for application that requires high I/O performance, one pod need a dedicated disk.
+* The bigdata application, e.g. Kafka, Cassandra, Hadoop can deal with a bunch of distinct volumes, one pod requires multiple disks.  
+* The emptyDir and hostPath volume doesn't have disk size limitation, the localDisk volume is able to provide this feature by using lvm.
+
+## Constraints and Assumptions
+
+1. The physical disks or logial volumes will be mounted to pre-defined mount-points in Nodes as below path format, and containers will use this host directory as data volume.
+  * disks, `/var/vol-pool/disks/<uuid>`
+  * logical volumes, `/var/vol-pool/lvm/<vg-name>/<lv-name>`
+
+2. If user requests a volume with type of `disk`, the whole disk will be assigned to container.

```

Comment:
From some of our users, they really care about I/O QoS. By using software solution, .e.g cgroup, it might be helpful, but no one is sure what will happen in the real production environment if multi-users are sharing same disk, and all of them have heavy I/O load.
For some cases, e.g. ceph, we even thought if we can expose disk device (/dev/sdX) to pod.